In [2]:
import pandas as pd
from Bio  import Seq

## Чтение исходной геномной таблицы

In [62]:
#чтение таблицы полученной из RASTEr.py

genome_tab = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/annotation_table/3817_genome_tab.csv')

## Добавление CAZy данных 
Есть две таблицы которы я качаю с dbCAN2:
* dbsub.out.tsv - Из этой таблицы стоит взять `dbCAN_subfam`, `Subfam_composition`, `substrate`, `E-Value`, `Coverage` 
* overview.tsv - `HMMER`, `DIMOND`, `Signalp`, `#ofTools`

In [63]:
dbsub_df = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/CAZy/dbsub.out.tsv', delimiter='\t')
overwiev_df = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/CAZy/overview.tsv', delimiter='\t')

In [117]:
dbsub_df.head()

dbCAN subfam                 Subfam Composition  \
0      GH1_e66                             GH1:92   
1      GH5_e70  GH5_29:45|GH5:1|GH5_27:1|GH5_54:1   
2     GH50_e17                             GH50:8   
3     GH5_e310                              GH5:7   
4    GH172_e10                          GH172:105   

                           Subfam EC                               Substrate  \
0  3.2.1.-:19|3.2.1.21:12|3.2.1.23:8  polyphenol, beta-glucan, beta-galactan   
1             3.2.1.123:1|3.2.1.46:1                  glycolipid, glycolipid   
2                                  -                                       -   
3                                  -                                       -   
4                                  -                                       -   

   Profile Length                 Gene ID  Gene Length        E Value  \
0             467  fig|1448933.6.peg.1062          474  4.600000e-223   
1             339  fig|1448933.6.peg.1161          439   1.100000e-42   
2             357  fig|1448933.6.peg.1169          365   3.000000e-66   
3             283  fig|1448933.6.peg.1185          604   2.700000e-33   
4             259  fig|1448933.6.peg.1237          361   9.800000e-99   

   Profile Start  Profile End  Gene Start  Gene End  Coverage  
0              1          466           3       454  0.995717  
1             33          332          74       331  0.882006  
2            168          357         173       358  0.529412  
3            113          283         119       286  0.600707  
4              1          258          65       330  0.992278

In [25]:
overwiev_df.head()

Gene ID                        EC#         HMMER dbCAN_sub  \
0  fig|1448933.6.peg.1062  3.2.1.-|3.2.1.21|3.2.1.23    GH1(2-455)   GH1_e66   
1  fig|1448933.6.peg.1161         3.2.1.123|3.2.1.46   GH5(52-335)   GH5_e70   
2  fig|1448933.6.peg.1169                          -  GH50(39-357)  GH50_e17   
3  fig|1448933.6.peg.1185                          -   GH5(40-285)  GH5_e310   
4  fig|1448933.6.peg.1187                          N  GH109(6-152)         N   

  DIAMOND  Signalp  #ofTools  
0     GH1        N         3  
1       N  Y(1-33)         2  
2       N        N         2  
3       N        N         2  
4       N        N         1

In [64]:

# Добавление данных из dbsub
for idx1, row1 in genome_tab.iterrows():
    for idx2, row2 in dbsub_df.iterrows():
        if row1['IDs'] == row2['Gene ID']:
            genome_tab.loc[idx1, 'dbCAN subfam'] = row2['dbCAN subfam']
            genome_tab.loc[idx1, 'Subfam Composition'] = row2['Subfam Composition']
            genome_tab.loc[idx1, 'Subfam EC'] = row2['Subfam EC']
            genome_tab.loc[idx1, 'E Value'] = row2['E Value']
            genome_tab.loc[idx1, 'Substrate'] = row2['Substrate']
            genome_tab.loc[idx1, 'Coverage'] = row2['Coverage']

In [65]:
# Добавление данных из overview
for idx1, row1 in genome_tab.iterrows():
    for idx2, row2 in overwiev_df.iterrows():
        if row1['IDs'] == row2['Gene ID']:
            genome_tab.loc[idx1, 'HMMER'] = row2['HMMER']
            genome_tab.loc[idx1, 'DIAMOND'] = row2['DIAMOND']
            genome_tab.loc[idx1, '#ofTools'] = row2['#ofTools']


In [121]:
genome_tab.head()

IDs             Description Type  Lenght_AA  \
0  fig|1448933.6.peg.26    hypothetical protein  CDS        386   
1  fig|1448933.6.peg.27    hypothetical protein  CDS         46   
2  fig|1448933.6.peg.28    hypothetical protein  CDS        582   
3  fig|1448933.6.peg.29  Deoxyhypusine synthase  CDS        364   
4  fig|1448933.6.peg.30    hypothetical protein  CDS         58   

                                                 AAs dbCAN subfam  \
0  MRWRLSAPFKAVVMGKGENTNLFREVMYFNGLSKEDQLRVIDALEG...          NaN   
1     MRRLFDLCVKVALRIVRWVWHPPLTSLYLEDEISSSRTATVKLIVC          NaN   
2  MGSFWQVNTLNSLPNRGAGLKNETSIWLRNALHKWIFIPLFMGSVI...          NaN   
3  MVEIKEVANLTPEERELKKELLSKPVEAIDPRSIETVSDLLHAYQN...          NaN   
4  MRCIKPSHPYEGIDAGVASLSLLTDFFLENPRLPMQGFTVECRPWE...          NaN   

  Subfam Composition Subfam EC  E Value Substrate  Coverage HMMER DIAMOND  \
0                NaN       NaN      NaN       NaN       NaN   NaN     NaN   
1                NaN       NaN      NaN       NaN       NaN   NaN     NaN   
2                NaN       NaN      NaN       NaN       NaN   NaN     NaN   
3                NaN       NaN      NaN       NaN       NaN   NaN     NaN   
4                NaN       NaN      NaN       NaN       NaN   NaN     NaN   

   #ofTools  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [66]:
genome_tab.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/CAZy/Geo3817_CAZy_only.tsv', index=False)

## Объединение таблиц MEROPS  и добавление в таблицу данных из нее
Так как я испольщуб EBI, там есть ограничение на колличестов последователньостей которые за раз он будет проганять для поиска по БД. По этому получается несколько аутпутов, кторые нужно обединить в одну таблицу

In [1]:
import pandas as pd

In [67]:
tab_p1 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/MEROPS/results/results_hmm_part_1.tsv', delimiter='\t')
tab_p2 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/MEROPS/results/results_hmm_part_2.tsv', delimiter='\t')
tab_p3 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/MEROPS/results/results_hmm_part_3.tsv', delimiter='\t')
tab_p4 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/MEROPS/results/results_hmm_part_4.tsv', delimiter='\t')

In [68]:
merops_tab = pd.concat([tab_p1, tab_p2, tab_p3, tab_p4])

In [69]:
merops_tab.columns

Index(['Sequence Number ', 'Query Name ', 'Hits Found ', 'Status ',
       'Top Hit  Identifier  ', 'Description ', 'E-value', 'Details',
       'Top Hit Identifier  '],
      dtype='object')

In [70]:
merops_tab = merops_tab[merops_tab['Hits Found '] > 0]

In [71]:
merops_tab.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/MEROPS/Geo3817_MEROPS_only.tsv', index=False)

In [72]:
merops_tab.head()

Sequence Number             Query Name   Hits Found  Status   \
2                  3  fig|1448933.6.peg.28             2   DONE    
40                41  fig|1448933.6.peg.66          3092   DONE    
58                59  fig|1448933.6.peg.84             1   DONE    
64                65  fig|1448933.6.peg.90         12368   DONE    
66                67  fig|1448933.6.peg.92             1   DONE    

   Top Hit  Identifier                                Description   \
2            MER0044154       family M50 non-peptidase homologues    
40           MER0065772                            CTP synthetase    
58           MER1152091   subfamily M23B non-peptidase homologues    
64           MER1360526       subfamily T1A unassigned peptidases    
66           MER0849883         family M20D unassigned peptidases    

          E-value Details Top Hit Identifier    
2    3.800000e-13    show                  NaN  
40  1.600000e-187    show                  NaN  
58   4.400000e-09    show                  NaN  
64  1.800000e-111    show                  NaN  
66   8.100000e-03    show                  NaN

In [128]:
genome_tab_caz = genome_tab[genome_tab['dbCAN subfam'].notna()]

In [129]:
#проверка какие белки по двум база проаннотировались чтобы узнать как будут следующие ячейки работать
for idx1, row1 in genome_tab_caz.iterrows():
    for idx2, row2 in merops_tab.iterrows():
        if row1['IDs'] == row2['Query Name '].strip():
            print(row1['IDs'])

fig|1448933.6.peg.372
fig|1448933.6.peg.675
fig|1448933.6.peg.1247
fig|1448933.6.peg.1299
fig|1448933.6.peg.2


fig|1448933.6.peg.372

fig|1448933.6.peg.675

fig|1448933.6.peg.1247


fig|1448933.6.peg.1299


fig|1448933.6.peg.2

In [73]:
copy = genome_tab.copy()

In [74]:
genome_tab.loc[1501, 'dbCAN subfam']

'GH3_e88'

In [54]:
type(genome_tab.loc[1501, 'dbCAN subfam'])

str

In [75]:
# Надо как то сделать проверку чтобы если в строке ужде есть что-то то он не перезаписывал а добавлял
for idx1, row1 in genome_tab.iterrows():
    for idx2, row2 in merops_tab.iterrows():
        if row1['IDs'] == (row2['Query Name ']).strip():
            if isinstance(row1['dbCAN subfam'], str):
                genome_tab.loc[idx1, 'dbCAN subfam'] = '/'.join([str(genome_tab.loc[idx1, 'dbCAN subfam']),str(row2['Top Hit  Identifier  ']) ])
                genome_tab.loc[idx1, 'E Value'] = '/'.join([str(genome_tab.loc[idx1, 'E Value']) , str(row2['E-value'])])
                genome_tab.loc[idx1, 'Substrate'] = '/'.join([str(genome_tab.loc[idx1, 'Substrate']), str(row2['Description '])])
            else:
                genome_tab.loc[idx1, 'dbCAN subfam'] = row2['Top Hit  Identifier  ']
                genome_tab.loc[idx1, 'E Value'] = row2['E-value']
                genome_tab.loc[idx1, 'Substrate'] = row2['Description ']


In [76]:
genome_tab

IDs             Description Type  Lenght_AA  \
0     fig|1448933.6.peg.26    hypothetical protein  CDS        386   
1     fig|1448933.6.peg.27    hypothetical protein  CDS         46   
2     fig|1448933.6.peg.28    hypothetical protein  CDS        582   
3     fig|1448933.6.peg.29  Deoxyhypusine synthase  CDS        364   
4     fig|1448933.6.peg.30    hypothetical protein  CDS         58   
...                    ...                     ...  ...        ...   
1498  fig|1448933.6.peg.21    hypothetical protein  CDS         93   
1499  fig|1448933.6.peg.22    hypothetical protein  CDS        364   
1500  fig|1448933.6.peg.23    hypothetical protein  CDS         38   
1501  fig|1448933.6.peg.24        beta-glucosidase  CDS        440   
1502  fig|1448933.6.peg.25    hypothetical protein  CDS        208   

                                                    AAs dbCAN subfam  \
0     MRWRLSAPFKAVVMGKGENTNLFREVMYFNGLSKEDQLRVIDALEG...          NaN   
1        MRRLFDLCVKVALRIVRWVWHPPLTSLYLEDEISSSRTATVKLIVC          NaN   
2     MGSFWQVNTLNSLPNRGAGLKNETSIWLRNALHKWIFIPLFMGSVI...  MER0044154    
3     MVEIKEVANLTPEERELKKELLSKPVEAIDPRSIETVSDLLHAYQN...          NaN   
4     MRCIKPSHPYEGIDAGVASLSLLTDFFLENPRLPMQGFTVECRPWE...          NaN   
...                                                 ...          ...   
1498  MKSIGLILGTHGGITRDICMAIGKGLPGGKEMEMALDDTRLRIDLA...          NaN   
1499  MVNKVRIGFIGAGGIANAHYQRYKNIPDVEIVAASDVDSAALERAQ...     GH109_e7   
1500             MLLLLELLTMAGLSARNMDFILSLISFVDLLRFSGDRP          NaN   
1501  MTVEEKVAQIGAVRAISLTENGSFSEEKARKFLSHGIGFITRLTGD...      GH3_e88   
1502  MSIKNWEKEYDVGIVKRAVMKVHPGMPLKERINHIIYRLNTQKQIL...          NaN   

     Subfam Composition                          Subfam EC E Value  \
0                   NaN                                NaN     NaN   
1                   NaN                                NaN     NaN   
2                   NaN                                NaN     0.0   
3                   NaN                                NaN     NaN   
4                   NaN                                NaN     NaN   
...                 ...                                ...     ...   
1498                NaN                                NaN     NaN   
1499           GH109:31                                  -     0.0   
1500                NaN                                NaN     NaN   
1501            GH3:482  3.2.1.37:15|3.2.1.55:3|3.2.1.21:1     0.0   
1502                NaN                                NaN     NaN   

                                 Substrate  Coverage         HMMER DIAMOND  \
0                                      NaN       NaN           NaN     NaN   
1                                      NaN       NaN           NaN     NaN   
2     family M50 non-peptidase homologues        NaN           NaN     NaN   
3                                      NaN       NaN           NaN     NaN   
4                                      NaN       NaN           NaN     NaN   
...                                    ...       ...           ...     ...   
1498                                   NaN       NaN           NaN     NaN   
1499                                     -  0.376923  GH109(3-153)       N   
1500                                   NaN       NaN           NaN     NaN   
1501          xylan, arabinan, beta-glucan  0.982456   GH3(67-288)     GH3   
1502                                   NaN       NaN           NaN     NaN   

      #ofTools  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  
...        ...  
1498       NaN  
1499       2.0  
1500       NaN  
1501       3.0  
1502       NaN  

[1503 rows x 14 columns]

In [77]:
genome_tab.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/CAZy/Geo3817_CAZy_MEROPS.tsv', index=False)

In [3]:
genome_tab = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/CAZy/Geo3817_CAZy_MEROPS.tsv')

# DeepTMHMM read

In [78]:
deeptmhmm_data = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/DeepTMHMM/table_parsed_geo3817.tsv', delimiter='\t')

In [79]:
deeptmhmm_data.head()

Name  Type  \
0  fig|1448933.6.peg.1  GLOB   
1  fig|1448933.6.peg.2  GLOB   
2  fig|1448933.6.peg.3  GLOB   
3  fig|1448933.6.peg.4  GLOB   
4  fig|1448933.6.peg.5  GLOB   

                                                  AA TM_composition  
0  ERRKVTFRVPAEILSFYDQYMRQVVEEGEYAVEVGSSSEDVRLSGK...          i1-68  
1  MNTYERLDNGIRLKSGTETTEITFLSEDIVKISRYTSRKAPELAVT...         o1-725  
2  MTCLSASQDRVRRFAQRLPTLETQNKRSLLINGLFPVRSPRAFAGG...          i1-71  
3  MIRNPKTPLGLFYSELAKEARCFGPPNIVKPNETSGTLIPQVQKAP...         i1-287  
4  MSTDKRNAVAVLSGGPDSMSYAALKKSEGYDIYPIIFDYGQKGKKE...         i1-233

In [80]:
for idx1, row1 in genome_tab.iterrows():
    for idx2, row2 in deeptmhmm_data.iterrows():
        if row1['IDs'] == row2['Name']:
            genome_tab.loc[idx1, 'TM Type'] = row2['Type']
            genome_tab.loc[idx1, 'TM_composition'] = row2['TM_composition']

## Фильтрация исходной таблицы для того чтбы добавить оставить только нужное

In [81]:
genome_tab.head()

IDs             Description Type  Lenght_AA  \
0  fig|1448933.6.peg.26    hypothetical protein  CDS        386   
1  fig|1448933.6.peg.27    hypothetical protein  CDS         46   
2  fig|1448933.6.peg.28    hypothetical protein  CDS        582   
3  fig|1448933.6.peg.29  Deoxyhypusine synthase  CDS        364   
4  fig|1448933.6.peg.30    hypothetical protein  CDS         58   

                                                 AAs dbCAN subfam  \
0  MRWRLSAPFKAVVMGKGENTNLFREVMYFNGLSKEDQLRVIDALEG...          NaN   
1     MRRLFDLCVKVALRIVRWVWHPPLTSLYLEDEISSSRTATVKLIVC          NaN   
2  MGSFWQVNTLNSLPNRGAGLKNETSIWLRNALHKWIFIPLFMGSVI...  MER0044154    
3  MVEIKEVANLTPEERELKKELLSKPVEAIDPRSIETVSDLLHAYQN...          NaN   
4  MRCIKPSHPYEGIDAGVASLSLLTDFFLENPRLPMQGFTVECRPWE...          NaN   

  Subfam Composition Subfam EC E Value                             Substrate  \
0                NaN       NaN     NaN                                   NaN   
1                NaN       NaN     NaN                                   NaN   
2                NaN       NaN     0.0  family M50 non-peptidase homologues    
3                NaN       NaN     NaN                                   NaN   
4                NaN       NaN     NaN                                   NaN   

   Coverage HMMER DIAMOND  #ofTools TM Type  \
0       NaN   NaN     NaN       NaN    GLOB   
1       NaN   NaN     NaN       NaN    GLOB   
2       NaN   NaN     NaN       NaN      TM   
3       NaN   NaN     NaN       NaN    GLOB   
4       NaN   NaN     NaN       NaN    GLOB   

                                      TM_composition  
0                                             i1-386  
1                                              i1-46  
2  i1-36 t37-57 o58-72 t73-83 i84-203 t204-221 o2...  
3                                             i1-364  
4                                              i1-58

In [150]:
genome_tab.columns

Index(['IDs', 'Description', 'Type', 'Lenght_AA', 'AAs', 'dbCAN subfam',
       'Subfam Composition', 'Subfam EC', 'E Value', 'Substrate', 'Coverage',
       'HMMER', 'DIAMOND', '#ofTools', 'TM Type', 'TM_composition'],
      dtype='object')

In [10]:
pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/Geo_table.csv').columns

Index(['Gene id', 'Description', 'Type', 'TM type', 'Lenght', 'AAs',
       'Protein Structure'],
      dtype='object')

In [82]:
genome_tab2 = genome_tab[['IDs','Description','Type','TM Type','Lenght_AA','AAs','TM_composition','dbCAN subfam','Subfam Composition','Substrate','E Value',
                         'Coverage','Subfam EC','HMMER','DIAMOND','#ofTools']]

In [20]:
genome_tab2.loc[1480, 'Lenght_AA'] = 68

In [84]:
genome_tab2.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/geo3817_full_table.csv', index=False)

# Одельные таблицы для CAZy Merops TM

In [85]:
geo3817_full_tab = pd.read_csv('/home/aster/Downloads/geo3817_full_table - geo3817_full_table.csv')

In [87]:
geo3817_full_tab.columns

Index(['IDs', 'Description', 'Type', 'TM Type', 'Lenght_AA', 'AAs',
       'TM_composition', 'dbCAN subfam', 'Subfam Composition', 'Substrate',
       'E Value', 'Coverage', 'Subfam EC', 'HMMER', 'DIAMOND', '#ofTools'],
      dtype='object')

In [127]:
Cazy_merops = geo3817_full_tab[(geo3817_full_tab['dbCAN subfam'].notna() & geo3817_full_tab['Substrate'].notna())]

In [128]:
Cazy_tab = Cazy_merops[~Cazy_merops['dbCAN subfam'].str.startswith('M')]

In [140]:
Cazy_tab.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/geo3817_cazy_only.csv', index=False)

In [131]:
MEROPS_tab = geo3817_full_tab[geo3817_full_tab['Substrate'].str.len() > 2]

In [138]:
MEROPS_tab2 = MEROPS_tab[~MEROPS_tab['dbCAN subfam'].str.contains('G', na=False)]

In [141]:
MEROPS_tab2.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/geo3817_peptidases_only.csv', index=False)

In [ ]:
geo3817_full_tab

In [147]:
TM_tab = geo3817_full_tab[geo3817_full_tab['TM Type'].str.contains('TM', na=False)]

In [148]:
TM_tab

IDs           Description Type TM Type  Lenght_AA  \
5        fig|1448933.6.peg.6  hypothetical protein  CDS      TM        236   
6        fig|1448933.6.peg.7  hypothetical protein  CDS      TM         81   
8        fig|1448933.6.peg.9  hypothetical protein  CDS      TM        445   
11      fig|1448933.6.peg.12  hypothetical protein  CDS      TM        144   
14      fig|1448933.6.peg.15  hypothetical protein  CDS      TM         62   
...                      ...                   ...  ...     ...        ...   
1474  fig|1448933.6.peg.1475  hypothetical protein  CDS      TM        116   
1476  fig|1448933.6.peg.1477  hypothetical protein  CDS      TM        383   
1480  fig|1448933.6.peg.1481  hypothetical protein  CDS      TM        281   
1481  fig|1448933.6.peg.1482  hypothetical protein  CDS      TM        285   
1482  fig|1448933.6.peg.1483  hypothetical protein  CDS      TM        277   

                                                    AAs  \
5     MTWEFYVWLCIIAFDVLGSAIWIRKFNNFILPLANYVLLVLMAQIF...   
6     MQRQIFIMGLALFITGLVMTLLTLNQIADISGAYYEYPFLTQQAYY...   
8     MGKNWYSQLPKGAKPALWSLFLGWMMDVYDVYLVAFLGLIIARVYL...   
11    MEYPRTAYILVLVGSVFQLLAALFETIGITAFFYVSPAPSRTYPAH...   
14    MQQVGVDYFSVGLFAAFFLGHSGIFDRRIRKGVVLAQAFLIIPQFL...   
...                                                 ...   
1474  MSENKQNQSGSQGKKPISTLLIAVIVIVIVVIGAVGYVAYQHYYVP...   
1476  MMRSQLILAVIFIVLITFAFRASNNMMVTTVPLVAKYDFDLSATEI...   
1480  MDGYKKVLTYVILIILAVIWLIPAYSMAVDGFKSNFDVLITPVLMP...   
1481  MDKQDHLLVVPIGITAAILIYLVAWNVYISLTNWSLLNPVPKVTWL...   
1482  MFIVRRKRSTSGNDESYTSKYLHYAFVISILALVVSAVGIFNGYLR...   

                                         TM_composition dbCAN subfam  \
5     o1-2 t3-23 i24-29 t30-47 o48-59 t60-78 i79-85 ...          NaN   
6                       i1-4 t5-24 o25-51 t52-70 i71-81          NaN   
8     i1-19 t20-42 o43-57 t58-80 i81-91 t92-112 o113...          NaN   
11    i1-13 t14-29 o30-60 t61-81 i82-87 t88-99 o100-...          NaN   
14                      o1-5 t6-19 i20-32 t33-43 o44-62          NaN   
...                                                 ...          ...   
1474                               i1-18 t19-39 o40-116          NaN   
1476  i1-10 t11-31 o32-45 t46-63 i64-73 t74-92 o93-9...          NaN   
1480  i1-6 t7-27 o28-65 t66-88 i89-103 t104-119 o120...          NaN   
1481  i1-6 t7-27 o28-61 t62-87 i88-97 t98-119 o120-1...          NaN   
1482  i1-20 t21-41 o42-69 t70-84 i85-102 t103-112 o1...          NaN   

     Subfam Composition                         Substrate  E Value  Coverage  \
5                   NaN                               NaN      NaN       NaN   
6                   NaN                               NaN      NaN       NaN   
8                   NaN  family M20 unassigned peptidases  2.3e-09       NaN   
11                  NaN                               NaN      NaN       NaN   
14                  NaN                               NaN      NaN       NaN   
...                 ...                               ...      ...       ...   
1474                NaN                               NaN      NaN       NaN   
1476                NaN                               NaN      NaN       NaN   
1480                NaN                               NaN      NaN       NaN   
1481                NaN                               NaN      NaN       NaN   
1482                NaN                               NaN      NaN       NaN   

     Subfam EC HMMER DIAMOND  #ofTools  
5          NaN   NaN     NaN       NaN  
6          NaN   NaN     NaN       NaN  
8          NaN   NaN     NaN       NaN  
11         NaN   NaN     NaN       NaN  
14         NaN   NaN     NaN       NaN  
...        ...   ...     ...       ...  
1474       NaN   NaN     NaN       NaN  
1476       NaN   NaN     NaN       NaN  
1480       NaN   NaN     NaN       NaN  
1481       NaN   NaN     NaN       NaN  
1482       NaN   NaN     NaN       NaN  

[275 rows x 16 columns]

In [149]:
TM_tab.to_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/geo3817_TM_only.csv', index=False)